src/back/spv/writer.rs

```diff
@@ -743,13 +792,13 @@ impl Writer {
                     crate::ScalarKind::Float => self.instruction_type_float(id, width),
                     crate::ScalarKind::Bool => self.instruction_type_bool(id),
                 };
-                self.lookup_type.insert(id, handle);
+                self.lookup_type.insert(id, lookup_type);
             }
             crate::TypeInner::Vector { size, kind, width } => {
                 let scalar_handle = self.find_scalar_handle(arena, kind, width);
-                let scalar_id = self.get_type_id(arena, scalar_handle);
+                let scalar_id = self.get_type_id(arena, &LookupType::Handle(scalar_handle));
                 instruction = self.instruction_type_vector(id, scalar_id, size);
-                self.lookup_type.insert(id, handle);
+                self.lookup_type.insert(id, lookup_type);
```

Comment:
This still gives a compile error, because it expects a value type instead of the reference type. Problem is if I dereference it, I get the error: "Cannot move".

Reply:
it looks like you are doing this `self.lookup_type.insert(id, lookup_type);` pretty much uniformly in this function.
Could it be moved outside of `match`?
And then, since this function needs it by value, we likely need to change the parameter to be passed by value

Reply:
That doesn't work because you don't have ownership over lookup_type, only a shared reference. Handles implement the Copy trait which is why it worked before, because it automatically copied the value.

While I haven't looked much at the code it looks like you actually do need to copy the lookup type here, but that would require `crate::Type` to be Clone and that you write `LookupType<T: Clone>` everywhere, which is messy. All the problems stem from the `LocalType::Local(T)` part, so I would look at another way to represent that.

Reply:
Did you consider `Local(&'a T)` instead?

Reply:
I haven't tried using a lifetime yet, I am still exploring the options. But I indeed stumbled across having to implement Clone for all generics that will be using this.

Reply:
Let's try to avoid IR changes for this an see if `&'a T` would be sufficient